src/Illuminate/Support/Url.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+
+class Url implements Arrayable
+{
+    /**
+     * Constructor.
+     */
+    public function __construct(
+        public $scheme = null,
+        public $host = null,
+        public $port = null,
+        public $user = null,
+        public $pass = null,
+        public $path = null,
+        public $query = null,
+        public $fragment = null,
```

Comment:
types here would be nice

Reply:
IIRC I think only objects and arrays are typed in the framework, but correct me if I'm wrong of course.

Reply:
Both property types and function return types are used numerously (property types in constructors a bit less though). I think you can add them, especially for new code (since there is no backwards-compat to keep in mind).

Reply:
Ok I've added types in 👍